File: src/Avalonia.Samples/CompleteApps/Avalonia.MusicStore/Messages/PurchaseAlbumMessage.cs
using Avalonia.MusicStore.ViewModels;
using CommunityToolkit.Mvvm.Messaging.Messages;

namespace Avalonia.MusicStore.Messages;

/// <summary>
/// Asynchronous request message to initiate the purchase of an album.
/// Sent via the messenger to request an <see cref="AlbumViewModel"/> from a handler,
/// typically when the user wants to add a new album to their collection.
/// </summary>
public class PurchaseAlbumMessage : AsyncRequestMessage<AlbumViewModel?>;

File: src/Avalonia.Samples/FSharp/CompleteApps/MusicStore/Messages/PurchaseAlbumMessage.fs
namespace MusicStore.Messages

open MusicStore.ViewModels
open CommunityToolkit.Mvvm.Messaging.Messages

module PurchaseAlbumMessage =
    type Command = { AlbumId: int; UserId: int }
    type Response = { Success: bool; Message: string option }

    /// Message used with CommunityToolkit's messenger to request an AlbumViewModel asynchronously.
    /// Receivers should call `message.Reply(Task<AlbumViewModel option>)`.
    type Self() =
        inherit AsyncRequestMessage<AlbumViewModel option>()
